### ISSUE TITLE
[Bug]: mo panic when run stability test

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#5334]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):0.6.0
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

2022/09/28 22:54:09.320858 +0800 INFO frontend/routine.go:142 connection id 1062 , the time of handling the request 1m28.347328377s
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1af539c]

goroutine 254 [running]:
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*TxnMVCCNode).ApplyCommit(0xc4e61d1f90, 0xc6c37e6bc0)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/txn/txnbase/mvccnode.go:297 +0x3c
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/catalog.(*TableMVCCNode).ApplyCommit(0xc313abbb10, 0xc6c37e6bc0?)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/catalog/tablemvccnode.go:72 +0x3b
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*MVCCChain).ApplyCommit(0xcaa26c4c60, 0xffffffffffffffff?)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/txn/txnbase/mvccchain.go:283 +0xb7
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnTable).ApplyCommit(0xcb53596a00)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/txn/txnimpl/table.go:734 +0xc9
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnDB).ApplyCommit(0xc974984b40)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/txn/txnimpl/txndb.go:357 +0x165
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnStore).ApplyCommit(0xc21695bbc0?)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/txn/txnimpl/store.go:455 +0x9e
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*Txn).ApplyCommit(0xc21695bc08?)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/txn/txnbase/txn.go:255 +0xa7
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*TxnManager).on1PCPrepared(0x234d030?, 0xc80048aee8)
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/txn/txnbase/txnmgr.go:286 +0x2ef
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*TxnManager).dequeuePrepared(0x418145?, {0xc0005da000, 0x2a, 0x0?})
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/txn/txnbase/txnmgr.go:370 +0x111
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/sm.(*safeQueue).Start.func1()
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/logstore/sm/safeq.go:83 +0x1f1
created by github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/sm.(*safeQueue).Start
        /data1/sudong/mo-regression/matrixone/pkg/vm/engine/tae/logstore/sm/safeq.go:62 +0xea

**Comment by LeftHandCold** [comment]
The root cause is that the front end did not deal with the data race, causing a txn to be in the commit stage and calling rollback at the same time.

**Comment by LeftHandCold** [comment]
![image](https://user-images.githubusercontent.com/14086886/193078209-d642a83f-a98a-4f2d-bba8-e037bc0b0d8a.png)
![image](https://user-images.githubusercontent.com/14086886/193078250-7b55d921-bcd1-4aa9-93d9-eeb75bf50805.png)

**Comment by LeftHandCold** [comment]
Now tae has a simple idempotent mechanism inside

**Comment by aressu1985** [comment]
it is a problem of stability test, and I trace this type of issue every daoy , when stability is OK , I will process all of these issues.

**Comment by aressu1985** [comment]
tonight will run the last time ,and if not, will close this issue

**Comment by aressu1985** [comment]
fixed: b9c225f4f44dcd62d6f577394d705b6aed55188e
